In [1]:
import pandas as pd
from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm
from statistics import mean
from scipy.stats import t

### Problem 14-61, P666-667
Jensen Tire & Auto is in the process of deciding whether to purchase a maintenance contract for its new computer wheel alignment and balancing machine. Managers feel that maintenance expense should be related to usage, and they collected the following information on weekly usage (hours) and annual maintenance expense (in hundreds of dollars).

|Weekely Useage in Hours|Annual Maintenance Expense in Hundreds of Dollars|
|:-:|:-:|
|13|17.0|
|10|22.0|
|20|30.0|
|28|37.0|
|32|47.0|
|17|30.5|
|24|32.5|
|31|39.0|
|40|51.5|
|38|40.0|


In [17]:
# a. Use ols module of statsmodels to fit the simple linear regression model 
# that related annual maintenance expense to weekely useage.
# Generate the summary of the regression results. (2 points)

# Prepare data
#X1 = [13, 10, 20, 28, 32, 17, 24, 31, 40, 38]
#y = [17.0, 22.0, 30.0, 37.0, 47.0, 30.5, 32.5, 39.0, 51.5, 40.0]
X1 = [10, 13, 17, 20, 24, 28, 31, 32, 38, 40]
y = [22.0, 17.0, 30.5, 30.0, 32.5, 37.0, 39.0, 47.0, 40.0, 51.5]

# Convert original data into dataframe that can be used by ols of statsmodel
data = pd.DataFrame({'X1':X1, 'y':y})

# Use ols module of statsmodels to fit the simple linear regression model
# Then generate the summary of the regression result.
model = ols('y~X1', data).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.856
Model:                            OLS   Adj. R-squared:                  0.838
Method:                 Least Squares   F-statistic:                     47.62
Date:                Sun, 03 Dec 2017   Prob (F-statistic):           0.000124
Time:                        21:48:41   Log-Likelihood:                -27.542
No. Observations:                  10   AIC:                             59.08
Df Residuals:                       8   BIC:                             59.69
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     10.5280      3.745      2.811      0.0

C:\Users\Ramanan-PC\Anaconda3\lib\site-packages\scipy\stats\stats.py:1334: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=10
  "anyway, n=%i" % int(n))


##### b. Clearly specify the simple linear regression model that related annual maintenance expense to weekely useage. Provide a practical intepretation of the coefficient of weekly usage. (2 points)

For an increase of 1 hour of weekly usage of balancing machine, the mean annual maintenance expense identified increases by 0.9534 hundreds of dollars i.e., 95.34 dollars


###### c. Provide a practical intepretation for the value of R-squared. (2 points)

85.6% of variation in the annual maintenance expenses can be explained by the variation in the weekly usage hours.

In [18]:
# d. Generate the corresponding ANOVA result. (2 points)
anova_results = anova_lm(model)
print(anova_results)


           df      sum_sq     mean_sq          F    PR(>F)
X1        1.0  860.050949  860.050949  47.623829  0.000124
Residual  8.0  144.474051   18.059256        NaN       NaN


C:\Users\Ramanan-PC\Anaconda3\lib\site-packages\scipy\stats\_distn_infrastructure.py:879: RuntimeWarning: invalid value encountered in greater
  return (self.a < x) & (x < self.b)
C:\Users\Ramanan-PC\Anaconda3\lib\site-packages\scipy\stats\_distn_infrastructure.py:879: RuntimeWarning: invalid value encountered in less
  return (self.a < x) & (x < self.b)
C:\Users\Ramanan-PC\Anaconda3\lib\site-packages\scipy\stats\_distn_infrastructure.py:1818: RuntimeWarning: invalid value encountered in less_equal
  cond2 = cond0 & (x <= self.a)


###### e. At the 5% level of significance, is there a significant relationship between the two variables? Why or why not? (2 points)

p-value from anova results = 0.000124

p-value < 0.05 (5% significance level), so YES, there is a significant relationship between the two variables.


In [19]:
# f. Provide the point estimate of the annual maintenance expense if the company expects to use the machine 30 hours per week.
# (2 points)
X1n = [30]
model.predict(exog=dict(X1=X1n))


0    39.13117
dtype: float64

In [24]:
# g. Compute the standard error of estimate. 
# Compute the the estimate of the standard deviation for the prediction interval of the annual maintenance expense
# given the company uses the machine 30 hours per week.
# (3 points)
n = 10
s = 18.059256**0.5
print("Standard error of estimate:", s)

# Compute the average weekly usage based on the sample data
X1bar = mean(X1)
#print(X1bar)

# Compute (n-1)*sample variance of X
svar = sum([(x-X1bar)**2 for x in X1])
#print(svar)

# Compute the estimate of the standard deviation for 30 hours usage per week
s_ci = s*pow(1/n + (30-X1bar)**2/svar, 0.5)
print("Estimate of Standard deviation for 30 hours usage per week:", s_ci)

Standard error of estimate: 4.249618335803817
Estimate of Standard deviation for 30 hours usage per week: 1.4925085715564712


In [26]:
# h. Develop a 95% prediction interval for the company’s annual maintenance expense 
# if the company expects to use the machine 30 hours per week.
# (3 points)
moe = t.ppf(0.5+0.95/2, 8)*s_ci    # The df is the degree of freedom of residual or error.
#print(moe)
#39.13117 is the point estimate of mean annual maintenance expense calculated above, based on 30 hours of usage per week
print(39.13117-moe, 39.13117+moe)


35.6894390624 42.5729009376


###### i. Please provide a practical interpretation of the prediction interval in question h. (2 points)

If the machine usage is 30 hours per week, we are 95% confident that the MEAN annual maintenance expense (in hundreds of dollars) will be between 35.6894390624 and 42.5729009376.